### PR TITLE
Week categories

### PR DESCRIPTION
**Pull request by kwood98**
Tests written for weekCategory.component, weekCategoreList, and weekCategoryService

### CODE REVIEW

**Review by RevatureGentry — APPROVED** [review]

**Comment by rlayneorr** [comment]
Please resolve the "bug" that SonarCloud is indicating.

File: store/WeekCategoryReducer.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    switch (action.type) {
+        case Actions.WeekCategoryActions.AddWeekCategory:
+            newState.weekCategories = [...state.weekCategories, action.payload];
+            return newState;
+        case Actions.WeekCategoryActions.DeleteWeekCategory:
+            newState.weekCategories = state.weekCategories.filter(
+                (cat) => cat.categoryId !== action.payload.categoryID
+              );
+              return newState;
+        case Actions.WeekCategoryActions.GetWeekCategories:
```
Comment (RevatureGentry) [inline store/WeekCategoryReducer.ts:20]: This action type has a misleading name - the name implies a read, and it shares the same functionality as `Actions.WeekCategoryActions.ChangeWeekCategories`, but it overwrites the existing `weekCategories` in state with the action's payload. Is this intentional?

File: weekCategories/WeekCategoryComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
+
+export default function CategoryButton(props:any) {
+    var [categoryContext,setCategoryContext] = useState(props.value);
+   return(
+    <View style={[styles.screenContainer] }>
+    <Text style={styles.myFontColor}>{categoryContext+"  "}
```
Comment (RevatureGentry) [inline weekCategories/WeekCategoryComponent.tsx]: Do we have the two whitespaces here to horizontally align the `categoryContext` with the `x` button?

**Review by RevatureGentry — CHANGES_REQUESTED** [review]
Consider changing the current action type of `Actions.WeekCategoryActions.GetWeekCategories` into a more appropriate name for the operation.

**Review by RevatureGentry — APPROVED** [review]

**Review by RevatureGentry — APPROVED** [review]

**Pull request merged**